src/main/java/com/wynntils/modules/utilities/overlays/OverlayEvents.java

```diff
@@ -677,27 +677,54 @@ public void onPlayerDeath(GameEvent.PlayerDeath e) {
         ConsumableTimerOverlay.clearConsumables(false);
     }
 
+
+    static boolean isVanished = false;
     @SubscribeEvent
-    public void onEffectApplied(PacketEvent<SPacketEntityEffect> e) {
+    public void onEffectApplied(PacketEvent<SPacketEntityEffect> e)
+    {
         if (!Reference.onWorld || !OverlayConfig.ConsumableTimer.INSTANCE.showSpellEffects) return;
 
+        // no idea what this does
         SPacketEntityEffect effect = e.getPacket();
         Potion potion = Potion.getPotionById(effect.getEffectId());
+
+        // if effect belongs to player?
         if (effect.getEntityId() == Minecraft.getMinecraft().player.getEntityId()) {
-            String timerName;
-            if (potion == MobEffects.SPEED) {
+            String timerName = "";
+
+            // if the effect is speed timer is "Speed boost"
+            if (potion == MobEffects.SPEED && effect.getAmplifier() == 2) {
                 timerName = "Speed boost";
-            } else if (potion == MobEffects.RESISTANCE && effect.getAmplifier() == 0) {
-                timerName = "War Scream I";
-            } else if (potion == MobEffects.RESISTANCE && effect.getAmplifier() == 1) {
-                timerName = "War Scream II";
-            } else if (potion == MobEffects.RESISTANCE && effect.getAmplifier() == 2) {
-                timerName = "War Scream III";
+            }
+            // if the effect is invisibility timer is "Vanish"
+            else if (potion == MobEffects.INVISIBILITY && effect.getDuration() < 200) {
+                timerName = "Vanish";
+                isVanished = true;
+            }
+            // if the player isn't invisible (didn't use vanish)
+            else if (!isVanished) {
```

Comment:
"isVanished" is true if and only if the previous if condition tested true, so you can just turn this into a simple else
```suggestion
            else {
```

Reply:
No. Because each effect is resolved individuallly, without it you get both Vanish and War Scream (II).

Reply:
Hm. Ok, if you say...